public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx

```diff
@@ -535,19 +543,22 @@ export const QueryAndExpressionsStep = ({ editingExistingRule, onDataChange, mod
                 }}
               />
             </Field>
-            {mode === 'edit' && (
-              <SmartAlertTypeDetector
-                editingExistingRule={editingExistingRule}
-                queries={queries}
-                rulesSourcesWithRuler={rulesSourcesWithRuler}
-                onClickSwitch={onClickSwitch}
-              />
+            {hasDatasourcesForDataSourceManaged && mode === 'edit' && (
+              <>
```

Comment:
I'm not sure about checking `hasDatasourcesForDataSourceManaged`, as in this case where we are editing an existing cloud rule, hasDatasourcesForDataSourceManaged has to be true

Reply:
Sure, I guess in this case it should always make sense to show it, but in practice you _shouldn't_ be able to see the page unless there are data source-managed rules? 🤔